lib/ShareProvider.js

```diff
@@ -324,12 +329,12 @@ ShareProvider.prototype = {
         node.setAttribute("image", data.url("content/img/glyph-share-16.svg"));
         node.setAttribute("observes", "Social:PageShareOrMark");
 
-        CustomizableUI.addListener(this);
-        Services.obs.addObserver(this, "social:providers-changed", false);
```

Comment:
turns out that the `this` here is different from the `this` in `observe()`... So I changed them all to use `shareButton` object